src/active_stakers.rs

```diff
@@ -76,8 +83,22 @@ impl ActiveStakers {
         pubkeys
     }
 
-    pub fn leader_schedule(&self) -> LeaderSchedule {
-        LeaderSchedule::new(self.pubkeys())
+    pub fn leader_schedule(&self, bank: &Bank, len: u64) -> LeaderSchedule {
+        let stakes = self.stakes();
+        let pubkeys = self.pubkeys();
+        let mut seed = [0u8; 32];
+        seed.copy_from_slice(&bank.last_id().as_ref()[..32]);
```

Comment:
You sure that `[..32]` is needed?  That last_id is 32 bytes. I'd be nice to see some compiler assurance here that they're matched.

Reply:
In fact, better than a pubkey, how about a 32-byte array? Give the caller the option to seed it differently.